code/controllers/CommentingController.php

```diff
@@ -130,80 +130,70 @@ public function getFeed(SS_HTTPRequest $request) {
 	 * Deletes a given {@link Comment} via the URL.
 	 */
 	public function delete() {
-		if(!$this->checkSecurityToken($this->request)) {
-			return $this->httpError(400);
-		}
+		$comment = $this->getComment();
+		if(!$comment) return $this->httpError(404);
+		if(!$comment->canDelete()) return $this->httpError(403);
+		if(!$comment->getSecurityToken()->checkRequest($this->request)) return $this->httpError(400);
 
-		if(($comment = $this->getComment()) && $comment->canDelete()) {
-			$comment->delete();
-				
-			return ($this->request->isAjax()) ? true : $this->redirectBack();
-		}
+		$comment->delete();
 
-		return $this->httpError(404);
+		return $this->request->isAjax()
+			? true
+			: $this->redirectBack();
 	}
 
 	/**
 	 * Marks a given {@link Comment} as spam. Removes the comment from display
 	 */
 	public function spam() {
-		if(!$this->checkSecurityToken($this->request)) {
-			return $this->httpError(400);
-		}
-
 		$comment = $this->getComment();
+		if(!$comment) return $this->httpError(404);
+		if(!$comment->canEdit()) return $this->httpError(403);
+		if(!$comment->getSecurityToken()->checkRequest($this->request)) return $this->httpError(400);
+		
+		$comment->IsSpam = true;
+		$comment->Moderated = true;
+		$comment->write();
 
-		if(($comment = $this->getComment()) && $comment->canEdit()) {
-			$comment->IsSpam = true;
-			$comment->Moderated = true;
-			$comment->write();
-				
-			return ($this->request->isAjax()) ? $comment->renderWith('CommentsInterface_singlecomment') : $this->redirectBack();
-		}
-
-		return $this->httpError(404);
+		return $this->request->isAjax()
+			? $comment->renderWith('CommentsInterface_singlecomment')
+			: $this->redirectBack();
 	}
 
 	/**
 	 * Marks a given {@link Comment} as ham (not spam).
```

Comment:
It is :smile:

Excuse the wikipedia link, but its as good a source for "common terms" as any! http://en.wikipedia.org/wiki/Naive_Bayes_spam_filtering#Mathematical_foundation